dashboard/test/controllers/xhr_proxy_controller_test.rb

```diff
@@ -0,0 +1,128 @@
+require 'webmock/minitest'
+WebMock.disable_net_connect!(:allow_localhost => true)
+require_relative '../../../shared/test/spy_newrelic_agent'
+require 'test_helper'
+
+class XhrProxyControllerTest < ActionController::TestCase
+  XHR_REDIRECT_URI = 'https://www.wikipedia.org/bar/a1b2'
+  XHR_URI = 'https://www.wikipedia.org/foo?a=1&b=2'
+  XHR_DATA = '{"key1":"value1", "key2":2, "obj":{"x":3, "y":4}}'
+  XHR_CONTENT_TYPE = 'application/json'
+  CHANNEL_ID = $stub_encrypted_channel_id
+  BAD_CHANNEL_MSG = "XhrProxyController request with invalid channel_id"
+
+  test "should fetch proxied media with correct content type" do
+    stub_request(:get, XHR_URI).to_return(body: XHR_DATA, headers: {content_type: XHR_CONTENT_TYPE})
+    get :get, u: XHR_URI, c: CHANNEL_ID
+    assert_response :success
+    assert_equal XHR_CONTENT_TYPE, response.content_type
+    cache_control = response['Cache-Control']
+    assert cache_control =~ /public/i, 'Response should be publicly cacheable'
+    assert cache_control =~ /max-age=60/i, 'Response should expire in 1 minute'
+    assert_equal response['Content-Transfer-Encoding'], 'binary'
+    assert_equal response['Content-Disposition'], 'inline'
+    assert_equal XHR_DATA, response.body
+  end
+
+  test "should handle query parameters" do
+    url = 'https://www.googleapis.com/freebase/v1/search?query=ada&filter=(any%20type:/people/person%20type:/location/citytown)'
+    stub_request(:get, url).to_return(body: XHR_DATA, headers: {content_type: XHR_CONTENT_TYPE})
+    get :get, u: url, c: CHANNEL_ID
+    assert_response :success
+    assert_equal XHR_DATA, response.body
+  end
```

Comment:
Note: this test verifies the query string is intact, but failed to catch a very specific bug where the query string was appended twice (e.g. ?foo=1?foo=1), possibly due to the forgiving way stub_request works. This didn't seem worth spending more time debugging now, but seems worth including this note in the PR in case it somehow comes up in the future.